lib/services/dapi/core/DapiCore.js

```diff
@@ -65,19 +84,33 @@ class DapiCore extends NodeJsService {
     this.dapiClient = new DapiClient({
       seeds,
       port: this.options.getRpcPort(),
+      nativeGrpcPort: this.options.getCoreGrpcPort(),
     });
 
+    const healthClient = new HealthCheckClient(
+      `127.0.0.1:${this.options.getCoreGrpcPort()}`,
+      grpc.credentials.createInsecure(),
+    );
+
+    const checkHealth = util.promisify(healthClient.check).bind(healthClient);
+
+    const request = new HealthCheckRequest();
+    request.setService('org.dash.platform.dapi.Core');
+
     let starting = true;
     let numIterations = 100;
+    let response;
     while (starting) {
       try {
         const blockHeight = await this.dapiClient.getBestBlockHeight();
         if (blockHeight > -1) {
           starting = false;
         }
+        response = await checkHealth(request);
       } catch (error) {
         if (error.message === "DAPI RPC error: getBestBlockHeight: Error: No MNs in list. Can't connect to the network.") {
           starting = false;
+          response = await checkHealth(request);
```

Comment:
That's looks bad... Don't you want to have two separate loops in two separate functions (`initializeJsonRpc`, `initializeGrpc`) and then call these functions in parallel from `initialize` method?